d1lod/d1lod/sesame/store.py

```diff
@@ -1,150 +1,334 @@
+This is a repository manager for the d1lod service to manage the Virtuoso repositories.
+It contains the following functionality:
 
-    1. Get a list of the repositories
-    2. Create repositories
-    3. Delete repositories
-    
+1. CREATE operation Creating a new repository in stores that support empty repositories.
+2. DELETE operation Removing a repository and all of its contents.
+3. COPY operation Modifying a repository to contain a copy of another.
+4. MOVE  operation Moving all of the data from one repository into another.
+5. ADD operation Reproducing all data from one repository into another.
+6. Getting a list of all the repositories.
 
-Reference material:
-
-http://rdf4j.org/sesame/2.7/docs/system.docbook?view#The_Sesame_Server_REST_API
-http://docs.s4.ontotext.com/display/S4docs/Fully+Managed+Database#FullyManagedDatabase-cURL%28dataupload%29
 """
 
 import requests
+import logging
 
 
 class Store:
-    def __init__(self, host="localhost", port=8080):
+    def __init__(self, host, port, name, ns={}):
+        self.name = name
+        self.ns = ns
         self.host = host
-        self.port = str(port)
-
+        self.port = port
         self.endpoints = {
-            'protocol': 'http://%s:%s/openrdf-sesame/protocol' % (self.host, self.port),
-            'repositories': 'http://%s:%s/openrdf-sesame/repositories' % (self.host, self.port),
-            'createRepository': 'http://%s:%s/openrdf-workbench/repositories/NONE/create' % (self.host, self.port),
-            'deleteRepository': 'http://%s:%s/openrdf-workbench/repositories' % (self.host, self.port)
+            'sparql': 'http://%s:%s/sparql' % (self.host, self.port),
         }
 
+        # Set aside a Session object
+        # We use this when making HTTP requests to reduce overhead when making
+        # many requests during the lifetime of a Repository objects
+        self.session = requests.Session()
+
 
     def __str__(self):
-        return """%s:%s""" % (self.host, self.port)
+        return "Repository: '%s'" % self.name
 
 
-    def protocol(self):
-        """Returns the protocol version for the Sesame interface."""
+    def exists(self, repository_name):
+        """Execute a SPARQL QUERY against the Repository.
 
-        endpoint = self.endpoints['protocol']
+        Arguments:
+        ----------
 
-        r = requests.get(endpoint)
+        repository_name: str
+            the <URI> to of the repository to check its existence
 
-        if r.status_code != 200:
-            print "Failed to get protocol %s." % name
-            print r.text
+        Returns: A boolean value representing the existence of the Repository."""
+        query_string = "ASK WHERE { GRAPH <" + repository_name + "> { ?s ?p ?o } }"
+        sparqlResponse = self.query(query_string=query_string, accept='text/plain')
+        return sparqlResponse.content
 
-        return r.text
 
+    def create_repository(self, repository_name, silent=False):
```

Comment:
Since we're no longer targeting Sesame, can we move away from the name Repository which doesn't have meaning outside of Sesame? I think the new equivalent is "Graph" or "Context" and I think Graph is most appropriate here.

Reply:
Refactored Graph and Interface modules and moved them to the default package.
Commit: https://github.com/ec-geolink/d1lod/pull/48/commits/83ea088f4d336fb606422dc6e0117c7f108ed410